profiles/models.py

```diff
@@ -22,11 +23,74 @@ class Profile(db.Model):
     id = db.Column(db.String(ID_LENGTH), primary_key=True)
     name = db.Column(db.Text(), nullable=False)
     orcid = db.Column(db.String(19), unique=True)
+    email_addresses = db.relationship('EmailAddress', order_by='EmailAddress.position',
+                                      collection_class=ordering_list('position'),
+                                      cascade='all, delete-orphan', back_populates='profile')
 
     def __init__(self, profile_id: str, name: str, orcid: str = None) -> None:
         self.id = profile_id
         self.name = name
         self.orcid = orcid
 
+    def update_from_orcid_record(self, orcid_record: dict):
```

Comment:
not sure arbitrarily traversing the orcid data structures belong in the domain layer, could be extracted as a OrcidRecordUpdate Command class